horace_core/sqw/coord_transform/@aProjection/private/get_contrib_cell_ind_.m

```diff
@@ -1,48 +1,88 @@
 function  contrib_ind = get_contrib_cell_ind_(source_proj,...
     cur_axes_block,targ_proj,targ_axes_block)
-% return the indexes of cells, which may contain
-% the nodes, belonging to the target axes block
+% Return the indexes of cells, which may contain the nodes,
+% belonging to the target axes block by
+%
+% Transforming the target coordinate system into source coordinate system
+% and binning TCS nodes into source coordinate system bins.
+%
 %
 
 if isempty(targ_proj)
     targ_proj = source_proj.targ_proj;
 end
-% Get the hypercube, which equal to minimal cell of the current grid
-% described by axes_block class.
-ch_cube = cur_axes_block.get_axes_scales();
-% and convert it into the target coordinate system assuming cube originates
-% from the centre of coordinates of the target coordinate system.
 %
-% TODO: investigate, if this is the best solution:
-% Make it half of real size to ensure at least one current grid cell point
-% appears in every grid cell of the target grid
-trans_chcube = 0.5*source_proj.from_this_to_targ_coord(ch_cube);
-
-% get all nodes belonging to target axes block, doing the
-% binning with the bin size, slightly smaller then the current
-% lattice size
+% build bin edges for the target grid and bin centres for reference grid
 if source_proj.do_3D_transformation_
-    [bin_nodes,dEnodes] = targ_axes_block.get_bin_nodes(trans_chcube,'-3D','-halo');
-    bin_range = targ_axes_block.img_range;
-    [any_inside,e_inside] = axes_block.bins_in_1Drange(dEnodes,bin_range(:,4));
-    if ~any_inside
+    [targ_nodes,dEnodes] = targ_axes_block.get_bin_nodes('-3D','-ngrid','-halo');
+    [ch_grid,baseEdges]  = cur_axes_block.get_bin_nodes('-bin_center','-3D');
+    presence_mark = ones(size(dEnodes));
+    presence_mark = nullify_edges(presence_mark,size(dEnodes));    
+    nodes_near = interp1(dEnodes,presence_mark,baseEdges,'linear',0);
+    may_contribure = nodes_near>0;
+    if ~any(may_contribure)
         contrib_ind = [];
         return;
     end
 else
-    bin_nodes = targ_axes_block.get_bin_nodes(trans_chcube,'-halo');
+    targ_nodes = targ_axes_block.get_bin_nodes('-ngrid','-halo');
+    ch_grid = cur_axes_block.get_bin_nodes('-bin_center');
 end
-% convert these nodes to the coordinate system, described by
-% the existing projection
-nodes_here = targ_proj.from_this_to_targ_coord(bin_nodes);
-% bin target nodes on the current lattice and return numbers of bins
-% contributed into lattice.
-nbin_in_bin = cur_axes_block.bin_pixels(nodes_here);
-%
-% identify cell indexes containing nodes
+szn = size(targ_nodes{1});
+presence_mark = ones(szn);
```

Comment:
What does `presence_mark` mean?

Reply:
mark of presence?

Reply:
@oerc0122 @abuts defined as per comment on L.40 in function nullify edges is I think what Jacob wants to know. L.35 seems to be a former definition and may provide a better name for this variable

Reply:
description read and modified